### PR TITLE
Update f/e dependencies

### LINKED ISSUE #52: Dependencies update
It's that time again folks! deps upgrade for f/e 🎉

### PR DESCRIPTION
**Pull request by mark-prins**
Fixes #52 

Have left two packages:
- zustand out ( that's still an rc version )
- msw ( the `0.41.0` version gives the error: "require function is used in a way in which dependencies cannot be statically extracted" ) and I see no need to update and we're not using msw significantly atm anyway

## Testing
- [ ] Should compile and run without error
- [ ] Packages updated, no more dependabot alerts

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -64,14 +64,14 @@
     "eslint-plugin-jest-dom": "^4.0.1",
     "eslint-plugin-react": "^7.27.1",
     "i18n-unused": "^0.8.0",
-    "jest": "^27.4.5",
+    "jest": "^28.1.0",
```
Comment (CarlosNZ) [inline client/package.json:67]: When running tests now, I get the following error:
```
Test environment jest-environment-jsdom cannot be found. Make sure the testEnvironment configuration option points to an existing node module.
```

Can fix by adding `jest-environment-jsdom` to dev dependencies.

Comment (CarlosNZ) [reply client/package.json:67]: Though all tests fail, even after that. 😩

**Review by CarlosNZ — CHANGES_REQUESTED** [review]
Thanks for wrangling all this! 

Getting this error when trying to run `yarn storybook`:
```
info => Loading presets
/Users/carl/GitHub/open-msupply/client/node_modules/file-system-cache/lib/common/Util.js:66
            throw new Error(`Failed to read cache value at: ${path}. ${error.message}`);
```

One more suggestion, slightly unrelated: might be a good idea to add a new yarn script "compile-full" which runs the compile command without the `--since HEAD` option. Useful when trying to compile after someone has managed to commit (possibly illegally).

**Comment by mark-prins** [comment]
Thanks - I'm not getting that storybook error, though am getting a different one in `react-refresh-webpack-plugin`
Checking now

```
Cannot read properties of undefined (reading 'tapPromise')
```

**Comment by CarlosNZ** [comment]
Screen Shot 2022-05-31 at 9 22 40 AM

@mark-prins here is that loading error I showed you. Tried full reset of all node_modules folders, no joy.

**Comment by mark-prins** [comment]
@CarlosNZ and anyone else who wants to have a try? Would you mind re-re-viewing?

I've created a new issue #112 for the remaining problem. I think now this isn't a blocker - it's just the dev run of storybook which is broken already, and I'd like to merge this PR to get the deps updated

File: server/cert-s/openssl.conf [hunk 2]
```diff
@@ -0,0 +1,94 @@
+[ req ]
+default_bits        = 2048
+default_keyfile     = server-key.pem
+distinguished_name  = subject
+req_extensions      = req_ext
+x509_extensions     = x509_ext
+string_mask         = utf8only
```
Comment (CarlosNZ) [inline server/cert-s/openssl.conf:7]: Is this file meant to be in here?

**Review by CarlosNZ — APPROVED** [review]
LGTM -- compiles ✅ tests pass ✅ seems to be running well ✅

Thanks for wrangling all this. 
Just one query about a server file.

File: server/cert-s/openssl.conf (hunk 2 above)
Comment (mark-prins) [reply server/cert-s/openssl.conf:7]: oops. no, will pull it out. It's a handy conf for generating self-signed certs 🙄

**Pull request merged**